linera-views/src/collection_view.rs

```diff
@@ -513,6 +531,51 @@ where
     }
 }
 
+impl<C, W> ByteCollectionView<C, W>
+where
+    C: Context + Send,
+    ViewError: From<C::Error>,
+    W: HashableView<C> + Sync,
+{
+    /// Computes the hash of the view
+    async fn compute_hash_internal(&self) -> Result<<sha3::Sha3_256 as Hasher>::Output, ViewError> {
+        let mut hasher = sha3::Sha3_256::default();
+        let keys = self.keys().await?;
+        hasher.update_with_bcs_bytes(&keys.len())?;
+        let updates = self.updates.write().await;
+        for key in keys {
+            hasher.update_with_bytes(&key)?;
+            let hash = match updates.get(&key) {
+                Some(entry) => {
+                    let Update::Set(view) = entry else {
+                        unreachable!();
```

Comment:
The keys are obtained from the call to `keys(&self)`. Therefore we know that the key exists.
If it were `Update::Removed`, then it would not exist. Therefore, this scenario cannot occur.

Reply:
ok thanks.